language/move-prover/bytecode-to-boogie/src/prelude.bpl

```diff
@@ -299,52 +310,58 @@ function {:inline 1} Dereference(m: Memory, ref: Reference): Value {
 }
 
 // Checker whether sender account exists.
-function ExistsTxnSenderAccount(m: Memory, txn: Transaction): bool {
+function {:inline 1} ExistsTxnSenderAccount(m: Memory, txn: Transaction): bool {
    // TODO: need to verify whether this is the intended semantics. We assume right now
    //   we can identify sender account existence if there is any resource under the sender address.
-   (exists resource: TypeValue :: domain#Memory(m)[Global(resource, sender#Transaction_cons(txn))])
+   // (exists resource: TypeValue :: domain#Memory(m)[Global(resource, sender#Transaction(txn))])
+   true
+}
+
+// Returns sender address.
+function {:inline 1} TxnSenderAddress(txn: Transaction): Address {
+  sender#Transaction(txn)
 }
 
 
 // ============================================================================================
 // Instructions
 
 procedure {:inline 1} Exists(address: Value, t: TypeValue) returns (dst: Value)
-requires is#Address(address);
 {
+    assume is#Address(address);
```

Comment:
Sorry for jumping in out of nowhere, I was just randomly browsing through the code. But maybe doing:
```
free requires is#Address(address);
```
would be more appropriate here.

Reply:
That is interesting. Changing this would go all over the code, let me do this (after consultation also with Shaz) in a different PR.